cmd/medius/publish/docs.go

```diff
@@ -0,0 +1,67 @@
+package publish
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"kubevirt.io/containerdisks/pkg/quay"
+
+	"github.com/ghodss/yaml"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/docs"
+)
+
+func NewPublishDocsCommand(options *common.Options) *cobra.Command {
+	options.PublishDocsOptions = common.PublishDocsOptions{
+		TokenFile: "",
+	}
+
+	publishCmd := &cobra.Command{
+		Use:   "docs",
+		Short: "Synchronize container disk descriptions with quay.io",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			success := true
+			client := quay.NewQuayClient(options.PublishDocsOptions.TokenFile, "containerdisks")
+			for _, p := range common.Registry {
+				if !p.UseForDocs {
+					continue
+				}
+				log := logger(p.Artifact)
+				metadata := p.Artifact.Metadata()
+				vm := docs.BasicVirtualMachine(metadata.Name, "quay.io/containerdisks/"+metadata.Name+":"+metadata.Version, metadata.ExampleCloudInitPayload)
+				example, err := yaml.Marshal(&vm)
+				if err != nil {
+					return fmt.Errorf("error marshaling example for for %q: %v", metadata.Name, err)
+				}
+				var result bytes.Buffer
+				data := &docs.TemplateData{
+					Name:        metadata.Name,
+					Description: metadata.Description,
+					Example:     string(example),
+				}
+				if err := docs.Template().Execute(&result, data); err != nil {
+					return fmt.Errorf("error rendering template for %q: %v", metadata.Name, err)
+				}
+
+				log.Info("Updating description on quay.io")
+				if options.DryRun == false {
```

Comment:
I am unsure here, too, about if the code is modifiable to work with another registry.

Reply:
The whole docs publishing is quay specific. I did not consider other repos at this stage with their own apis and capabilities. I would want to defer such considerations.

Reply:
ok